### PR TITLE
Fix list_report third argument

### LINKED ISSUE #131: list_report can fail if no 3rd argument
Fix Line 137 of `mailing_list.py` (https://github.com/goavki/phenny/commit/d0246c22a9e058403a8d4355be4d7cfe911e7b64#diff-83eb0c0023d4524d35045cd0e95a6b01R137) so that it behaves correctly if there are 3 groups.

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #131
@shardulc Friendly ping.

### CODE REVIEW

File: modules/mailing_list.py [hunk 1]
```diff
@@ -122,26 +122,33 @@ def last_message(phenny, ml):
...
-        phenny.reply(syntax.format(input.group(1), ', '.join(phenny.config.mailing_lists.keys())))
+            if input.group(3):
+                if input.group(3) in phenny.config.mailing_lists:
+                    phenny.reply(last_message(phenny, input.group(3)))
+                    return
+            else:
+                for i in phenny.config.mailing_lists:
+                    phenny.reply(last_message(phenny, i))
+
+                return
```
Comment (wei2912) [inline modules/mailing_list.py]: Duplicated on line 142 and 147 across both branches.

File: modules/mailing_list.py [hunk 2]
```diff
@@ -122,26 +122,33 @@ def last_message(phenny, ml):
...
+
+    phenny.reply(syntax.format(input.group(1), ', '.join(phenny.config.mailing_lists.keys())))
+
 list_report.name = "mailing list reporter"
-#list_report.rule = r'.(mailinglist|ml)(?:\s(poll|last(?:\s([\w-]+))?))?'
-list_report.commands = ['ml', 'mailinglist']
```
Comment (wei2912) [inline modules/mailing_list.py]: Why was this line removed, and the next one uncommented?

Comment (Androbin) [reply modules/mailing_list.py]: See my comment here:
https://github.com/goavki/phenny/commit/d0246c22a9e058403a8d4355be4d7cfe911e7b64#commitcomment-26484146

File: modules/mailing_list.py (hunk 1 above)
Comment (Androbin) [reply modules/mailing_list.py]: No, the condition on line 140 may fail.

Comment (shardulc) [reply modules/mailing_list.py]: Add an `else` branch to line 140 for readability.

Comment (Androbin) [reply modules/mailing_list.py]: Shall I re-duplicate line 149?

Comment (shardulc) [reply modules/mailing_list.py]: Actually, I'm not sure I understand the flow here. Give me a minute.

**Comment by Androbin** [comment]
@shardulc Are you okay with the flag solution?

**Comment by shardulc** [comment]
@Androbin Yes, this looks better to me.

**Comment by Androbin** [comment]
@shardulc What is left to do for this task?

**Comment by shardulc** [comment]
@Androbin Probably not much, but I've been busy with college apps so sorry I couldn't properly review all the changes you made in these last couple of days. I'll try to catch up today.

**Comment by shardulc** [comment]
@jonorthwash I think this should now make the mailing list stuff work properly. But because this hasn't been tested with an actual mailing list configuration, this may have to be changed later. I'm merging it for now.
